libres/lib/analysis/ies/ies.cpp

```diff
@@ -416,3 +416,44 @@ void ies::initX(const config::Config &ies_config, const Eigen::MatrixXd &Y0,
                 ies_config.truncation(), use_aa_projection, W0, steplength,
                 iteration_nr, nullptr);
 }
+
+Eigen::MatrixXd ies::makeE(const Eigen::VectorXd &obs_errors,
+                           const Eigen::MatrixXd &noise) {
+    int active_obs_size = obs_errors.rows();
+    int active_ens_size = noise.cols();
+
+    Eigen::MatrixXd E = noise;
+    Eigen::VectorXd pert_mean = E.rowwise().mean();
+    E = E.colwise() - pert_mean;
+    Eigen::VectorXd pert_var = E.cwiseProduct(E).rowwise().sum();
+
+    for (int i = 0; i < active_obs_size; i++) {
```

Comment:
```cpp
Eigen::VectorXd factor = obs_errors.cwiseProduct(
    (active_ens_size * pert_var.cwiseInverse()).cwiseSqrt());
E = E.array().colwise() * factor.array();
```

Reply:
Not sure if this is better.
We are not that used to Eigen yet, so hard to tell...

Reply:
hard pass :laughing: